assets/app/scripts/controllers/javaLink.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:JavaLinkController
+ * @description
+ * # ProjectController
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .run(function(ProxyPod, BaseLink, HawtioExtension, $templateCache, $compile) {
+    var template =
+'<span ng-show="container.jolokiaUrl" title="View java details">' +
+'  <a ng-click="gotoContainerView(container, container.jolokiaUrl)" href="">' +
+'    <i class="fa fa-external-link"></i>' +
+'  </a>' +
+'</span>';
+    HawtioExtension.add('container-links', function ($scope) {
+      var pod = (UI.findParentWith($scope, 'pod') || {}).pod;
```

Comment:
hrmm... maybe, but this template is used in lots of other places... `findParentWith` just smells wrong... in the case where the podTemplate is associated with a single pod, you could just use the `podTemplate` scope var. That said, I'm wondering why we wouldn't render the link inside the actual pod objects